components/nav-avatar.tsx

```diff
@@ -0,0 +1,145 @@
+import { useState } from "react";
+import { OverlayTrigger, Popover } from "react-bootstrap";
+
+import CloseIcon from "assets/icons/close-icon";
+import ExternalLinkIcon from "assets/icons/external-link-icon";
+
+import Avatar from "components/avatar";
+import Identicon from "components/identicon";
+import InternalLink from "components/internal-link";
+
+import { useAuthentication } from "contexts/authentication";
+
+import { truncateAddress } from "helpers/truncate-address";
+
+import useNetworkTheme from "x-hooks/use-network";
+
+const Link = (label, href) => ({ label, href });
+
+const ProfileInternalLink = ({ label, href }) => 
```

Comment:
these templates should be inside the NavAvatar function as a closure, since they are not exported nor used anywhere else - no need for them to be outside of the scope that uses them